wolfcrypt/src/port/tropicsquare/tropic01.c

```diff
@@ -0,0 +1,491 @@
+                        info->cipher.aesgcm_dec.ivSz,
+                        info->cipher.aesgcm_dec.authTag,
+                        info->cipher.aesgcm_dec.authTagSz,
+                        info->cipher.aesgcm_dec.authIn,
+                        info->cipher.aesgcm_dec.authInSz);
+
+                    /* reset devId */
+                    info->cipher.aesgcm_dec.aes->devId = devId;
+                }
+            }
+#endif /* HAVE_AESGCM */
+    #ifdef HAVE_AES_CBC
+        if (info->cipher.type == WC_CIPHER_AES_CBC) {
+            if (info->cipher.enc) {
+
+                ret = Tropic01_GetKeyAES(
+                    info->cipher.aescbc.aes,
+                    TROPIC01_AES_RMEM_SLOT_DEFAULT,
+                    TROPIC01_AES_MAX_KEY_SIZE);
+                if (ret != 0) {
+                    WOLFSSL_MSG_EX(
+                        "TROPIC01: CryptoCB: Failed to get AES key, ret=%d",
+                        ret);
+                    return ret;
+                }
+                /* set devId to invalid, so software is used */
+                info->cipher.aescbc.aes->devId = INVALID_DEVID;
+                ret = wc_AesCbcEncrypt(
+                    info->cipher.aescbc.aes,
+                    info->cipher.aescbc.out,
+                    info->cipher.aescbc.in,
+                    info->cipher.aescbc.sz);
+
+                /* reset devId */
+                info->cipher.aescbc.aes->devId = devId;
+            }
+            else {
+
+                ret = Tropic01_GetKeyAES(
+                    info->cipher.aescbc.aes,
+                    TROPIC01_AES_RMEM_SLOT_DEFAULT,
+                    TROPIC01_AES_MAX_KEY_SIZE);
+                if (ret != 0) {
+                    WOLFSSL_MSG_EX(
+                        "TROPIC01: CryptoCB: Failed to get AES key, ret=%d",
+                         ret);
+                    return ret;
+                }
+                /* set devId to invalid, so software is used */
+                info->cipher.aescbc.aes->devId = INVALID_DEVID;
+                ret = wc_AesCbcDecrypt(
+                    info->cipher.aescbc.aes,
+                    info->cipher.aescbc.out,
+                    info->cipher.aescbc.in,
+                    info->cipher.aescbc.sz);
+
+                /* reset devId */
+                info->cipher.aescbc.aes->devId = devId;
+            }
+        }
+    #endif /* HAVE_AES_CBC */
+#endif /* !NO_AES || !NO_DES3 */
+            break;
+
+        default:
+
+            break;
+    }
+
+    return ret;
+}
+/* Set TROPIC01 pairing keys */
+int Tropic01_SetPairingKeys(int kIndex, const byte* kPub, const byte* kPriv)
+{
+
+    if (kPub == NULL || kPriv == NULL || kIndex < 0 || kIndex > 3) {
+        WOLFSSL_MSG_EX("TROPIC01: SetPairingKeys: Invalid arguments");
+        return BAD_FUNC_ARG;
+    }
+
+    WOLFSSL_MSG_EX(
+        "TROPIC01: SetPairingKeys: Setting pairing key in slot %d",
+        kIndex);
+
+    for (int i = 0; i < TROPIC01_PAIRING_KEY_SIZE; i++) {
```

Comment:
Portability issue "mixed declaration". Please move the `int i` to top of function.